### PR TITLE
tecmosys.cpp : Updates

### PR DESCRIPTION
**Pull request by cam900**
tecmosys.cpp : Add notes, Fix background color, Cleanup bankswitching, Reduce duplicates, Cliprect related mixing, Add object finders instead runtime tag lookup

Demote tecmosys.cpp games with MACHINE_IMPERFECT_GRAPHICS and MACHINE_NO_COCKTAIL, Because some graphic features(linescroll, mixing) is not perfectly emulated, And tilemap scroll is wrong when flip screen case

### CODE REVIEW

File: src/mame/video/tecmosys.cpp [hunk 1]
```diff
@@ -264,53 +204,62 @@ void tecmosys_state::do_final_mix(bitmap_rgb32 &bitmap)
 
 				dstptr[x] = b | (g<<8) | (r<<16);
 			}
-			else if (pri2 >= pri)
+			else if ((pri2 >= pri) && ((mask != 0x3f00) || (mask != 0)))
```
Comment (cuavas) [inline src/mame/video/tecmosys.cpp]: This doesn't make sense - the condition `((mask != 0x3f00) || (mask != 0))` cannot be false, as `mask` cannot be equal to both 0x3f00 and 0 at the same time.

**Comment by Osso13** [comment]
This builds fine with official tools.

**Comment by cuavas** [comment]
@cam900 thanks, I cleaned up the change a bit in 8faa36c7e1486d0b0b6b5b47c73ddebceee9b46c:
* In general lambdas are better than macros in cases like this, as scope behaves more naturally, and you get more useful error messages when things to wrong.
* It's generally good to keep scope tight and mark things `const` if they aren't supposed to change after initialisation.

**Comment by DavidHaywood** [comment]
This broke the background colour in deroon test mode, it's now bright pink instead of black.

**Pull request merged**